.github/workflows/versions-propagation-auto-merge.yml

```diff
@@ -0,0 +1,21 @@
+name: Versions propagation auto-merge
+on:
+  workflow_call:
+
+jobs:
+  enable-auto-merge:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'alfresco-build' && contains( github.event.pull_request.labels.*.name, 'updatebot') }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+      - name: Enable auto-merge for propagation Pull Request
+        run: gh pr merge --auto --squash "$PR_URL"
+        env:
+          GITHUB_TOKEN: ${{secrets.BOT_GITHUB_TOKEN}}
+      - name: Auto-approve propagation Pull Request
+        run: gh pr review --approve "$PR_URL"
+        env:
+          PR_URL: ${{github.event.pull_request.html_url}}
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
```

Comment:
This token is intentionally different from the one in `line 16`, as we need to have a different user from the one who created the PR to approve it: `github-actions` is approving the PR and `alfresco-build`  is enabling the auto-merge.

Reply:
:+1: Did not see your comment here before I commented on it below: imho this deserves be be stated as a comment in the code.

Reply:
I added comments on both tokens, explaining why which one is used.